packages/dds/tree/src/events/events.ts

```diff
@@ -97,47 +114,35 @@ export interface IEventEmitter<E extends Events<E>> {
  * Or, compose over it:
  * @example
  * ```ts
- * class MyClass extends IEventEmitter<MyEvents> {
+ * class MyClass implements ISubscribable<MyEvents> {
  *   private readonly events = EventEmitter.create<MyEvents>();
  *
  *   private load() {
  *     this.events.emit("loaded");
  *   }
  *
- *   public on<K extends (string | symbol) & keyof MyEvents>(
- *     eventName: K,
- *     listener: MyEvents[K],
- *   ): () => void {
- *     return events.on(eventName, listener);
+ *   public on<K extends keyof MyEvents>(eventName: K, listener: MyEvents[K]): () => void {
+ *     return this.events.on(eventName, listener);
  *   }
  * }
  * ```
  */
-export class EventEmitter<E extends Events<E>> implements IEventEmitter<E> {
-    private readonly listeners: Partial<{
-        [P in keyof E]: Set<E[P]>;
-    }> = {};
+export class EventEmitter<E extends Events<E>> implements ISubscribable<E> {
+    private readonly listeners = new Map<keyof E, Set<(...args: unknown[]) => void>>();
 
     /**
-     * Create an instance of an {@link EventEmitter}.
+     * Create an instance of an {@link EventEmitter} that can be told to emit events. This is useful
+     * for classes that wish to compose over an {@link EventEmitter} rather than inheriting from it.
      */
-    public static create<E extends Events<E>>() {
-        return new EventEmitter() as EventEmitter<E> & {
-            // Expose the `emit` method so that it may be called by the creator.
-            emit: EventEmitter<E>["emit"];
-        };
+    public static create<E extends Events<E>>(): EventEmitter<E> & IEmitter<E> {
+        return new ComposableEventEmitter<E>();
     }
 
-    // The constructor is private to seal the class as well as to require use of the static `create` function
-    private constructor() {}
+    // The constructor is protected so as to require use of the static `create` function
```

Comment:
I think we could say why this is the case in the comment instead of what is done here, especially since this is an implementation comment not a doc comment.
```suggestion
    // Calling this externally (not from a subclass) would make sending events to it impossible.
    // Instead use the static `create` function to get an instance which allows emitting events.
```